pkg/apis/event_stats.go

```diff
@@ -31,62 +65,121 @@ func (es EventStats) StoreEventData() gin.HandlerFunc {
 		// change this to call from a struct
 		if err := sd.StoreEventData(data); err != nil {
 			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+
+			ReqDetailCount.ErrorRequestCount = ReqDetailCount.ErrorRequestCount + 1
```

Comment:
Since all these metrics are very generic and we would like to have these for all the endpoints, rather than tightly integrating it with business logic better to create a middleware and add these monitoring logic there.

Reply:
updated the code to add middleware for monitoring